### DISCUSSION TITLE
Custom AzureAI deployment in virtual network - how to customize the endpoint base path?

### DISCUSSION THREAD

**Discussion by bodzebod** [FlowiseAI/Flowise#2710] [Q&A] [answered]
Hi, my company deployed internally, on an AzureAI virtual network, some OpenAI models. This is described on https://learn.microsoft.com/en-us/azure/api-management/api-management-using-with-internal-vnet?tabs=stv2 .

So the base URL is very different from public URL, with domain name ending with ".azure-api.net" . I've been unable to configure OpenAI chats agents due to the fact that I can't customize the endpoint base path for it.
Basically its URL looks like that: https://xxxxxxxxxx.azure-api.net/openai-test/openai/deployments/gpt-4o/chat/completions?api-version=2024-02-15-preview

Is there a way to make it in flowise or some plans to add support for that? Thank you!

**Comment by bodzebod** [comment, marked as answer]
Sorry, I finally found a similar question here: https://github.com/langchain-ai/langchainjs/issues/1272 .
Looks like there had a fix here: https://github.com/langchain-ai/langchainjs/pull/1792 .

**Reply by Wongkitlong314** [reply]
may I know how can I add the AZURE_OPENAI_ENDPOINT in flowise

**Reply by bodzebod** [reply]
it's an environment variable, it's not set inside flowise ui.
